modules/oe_theme_content_event/src/Plugin/ExtraField/Display/RegistrationButtonExtraField.php

```diff
@@ -99,12 +100,22 @@ public function viewElements(ContentEntityInterface $entity) {
     if ($event->isRegistrationPeriodYetToCome($this->requestDateTime)) {
       $datetime_start = $event->getRegistrationStartDate();
       $datetime_end = $event->getRegistrationEndDate();
-      $date_diff = $this->dateFormatter->formatDiff($this->requestTime, $datetime_start->getTimestamp());
-      $build['#description'] = $this->t('Registration will open in @time_left. You can register from @start_date, until @end_date.', [
-        '@time_left' => $date_diff,
-        '@start_date' => $this->dateFormatter->format($datetime_start->getTimestamp(), 'oe_event_date_hour'),
-        '@end_date' => $this->dateFormatter->format($datetime_end->getTimestamp(), 'oe_event_date_hour'),
-      ]);
+      $request_datetime = DrupalDateTime::createFromTimestamp($this->requestTime);
+      $date_diff = $request_datetime->diff($datetime_start);
+      if ($date_diff->days === 0) {
```

Comment:
This condition has a problem because if the time remaining is less than 24h it will trigger, even though it might be a different day.